control-plane/config/sink/100-kafka-sink.yaml

```diff
@@ -0,0 +1,140 @@
+# Copyright 2020 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: kafkasinks.eventing.knative.dev
+  labels:
+    duck.knative.dev/addressable: "true"
+    eventing.knative.dev/release: devel
+spec:
+  group: eventing.knative.dev
+  names:
+    kind: KafkaSink
+    plural: kafkasinks
+    singular: kafkasink
+  scope: Namespaced
+  versions:
+    - name: v1alpha1
+      served: true
+      storage: true
+      subresources:
+        status: { }
+      schema:
+        openAPIV3Schema:
+          description: 'Kafka Sink is Addressable, it receives events and send them to a Kafka topic.'
+          type: object
+          properties:
+            spec:
+              description: 'Spec defines the desired state of the Kafka Sink.'
+              type: object
+              required:
+                - topic
+                - numPartitions
+                - replicationFactor
+                - bootstrapServers
+              properties:
```

Comment:
I think we need an optional "encoding" flag here, to let the user choose between structured and binary. Also, i think the default should be structured (it's quite unnatural in kafka world to send stuff in headers, so i think kafka users would prefer the structured mode as default)

Reply:
CE spec calls them message `mode`: https://github.com/cloudevents/spec/blob/master/spec.md#message, do you think `encoding` it's more descriptive?

Reply:
To me, on this context, mode seems too "broad"... But i would love to hear people other people opinions @matzew

Reply:
/cc @lionelvillard @n3wscott

Reply:
Or `contentMode`

Reply:
+1 for `contentMode`. We also need a Sink duck type!

Reply:
Have you already talked about a Sink duck type? I've not found a related issue.
I would love to have input on how a Sink duck type would look like. 
I guess it must be Addressable, so `status.address.url` (and we have `AddressStatus` duck), but I'm not sure what other fields make a sink different from an `Addressable`.

Reply:
@slinkydeveloper Is it ok for you if we use `contentMode`?

Reply:
I'm not aware of any discussions around a Sink duck type but I'm sure @n3wscott has some ideas. As you said it must be addressable, and it should includes `ceOverrides`, `contentMode` and possibly `dataContentType` (For RedisStreamSink data can either be json-encoded or RESP-encoded). Anyway do you mind opening an issue?

Reply:
https://github.com/knative/eventing/issues/3973

Reply:
As yesterday discussion during eventing sources wg, I think for now we should proceed forward to shape this `KafkaSink` CRD as an `Addressable` and nothing else, since my understanding is that this is the only need for a Sink.
If at some point, we need to abstract something more than `Addressable` for a sink, we can create that sink duck type and bump this `KafkaSink` CRD. This process should also help us to better understand what we need to abstract, other than Addressable itself.
WDYT @n3wscott @lionelvillard @pierDipi ?

Reply:
I would not call it a Sink. The concept is a sink, just like a source is a source in concept. Call it what it is for Kafka is my recommendation, but I have had a lot of pushback on this idea. Folks just want to add the label to everything....

A kafka source would be KafkaSubscription
A kafka sink would be a KafkaTopic

A kafka channel would make n KafkaSubscription for the controlled KafkaTopic